sim/core/item_swaps.go

```diff
@@ -212,6 +212,40 @@ func (swap *ItemSwap) RegisterActive(itemID int32) {
 	})
 }
 
+// Helper for handling Enchant On Use effects to set a 30s cd on the related spell.
+// Currently only used for random suffix "enchants" like the Karazhan gloves.
+func (swap *ItemSwap) RegisterEnchantActive(effectID int32, spellID int32) {
```

Comment:
Would be nice to merge this with the function above and have a `registerActiveInternal` function that accepts a config to reuse this logic

Reply:
There's a few differences though, checking for an item vs checking for an effect but I guess that would be done with the config struct then?

Reply:
Yes, similar to the Proc config which can have an ItemID or a SpellID. If both of them are set you could do the `EnchantActive` specific logic